PurchasesCoreSwift/Identity/IdentityManager.swift

```diff
@@ -0,0 +1,163 @@
+            Logger.warn(Strings.identity.logging_in_with_same_appuserid)
+            purchaserInfoManager.purchaserInfo(appUserID: currentAppUserID) { maybePurchaserInfo, maybeError in
+                completion(maybePurchaserInfo, false, maybeError)
+            }
+            return
+        }
+
+        backend.logIn(currentAppUserID: currentAppUserID,
+                      newAppUserID: newAppUserID) { maybePurchaserInfo, created, maybeError in
+            if maybeError == nil,
+               let purchaserInfo = maybePurchaserInfo {
+                self.deviceCache.clearCaches(oldAppUserID: currentAppUserID, andSaveWithNewUserID: newAppUserID)
+                self.purchaserInfoManager.cache(purchaserInfo: purchaserInfo, appUserID: newAppUserID)
+            }
+
+            completion(maybePurchaserInfo, created, maybeError)
+        }
+    }
+
+    @objc public func logOut(completion: (Error?) -> Void) {
+        Logger.info(String(format: Strings.identity.log_out_called_for_user, maybeCurrentAppUserID ?? "<nil currentAppUserID>"))
+
+        if currentUserIsAnonymous {
+            completion(ErrorUtils.logOutAnonymousUserError())
+            return
+        }
+
+        resetAppUserID()
+        Logger.info(Strings.identity.log_out_success)
+        completion(nil)
+    }
+
+    // TODO (post-migration): Change back to private.
+    func generateRandomID() -> String {
+        "$RCAnonymousID:\(UUID().uuidString.replacingOccurrences(of: "-", with: "").lowercased())"
+    }
+
+}
+
+// MARK: Deprecated
+// TODO: Migrate off these so we can mark them deprecated.
+extension IdentityManager {
+
+    @objc(identifyAppUserID:completion:)
+    public func identify(appUserID: String, completion: @escaping (Error?) -> Void) {
+        // TODO: Old code assumed we weren't nil, but it doesn't actually look like we need a value since
+        // we'll end up changing the app user id anyway.
+        let currentAppUserID = maybeCurrentAppUserID ?? ""
+        if currentUserIsAnonymous {
+            Logger.user(String(format: Strings.identity.identifying_anon_id, currentAppUserID))
+            createAlias(appUserID: appUserID, completion: completion)
+        } else {
+            Logger.user(String(format: Strings.identity.changing_app_user_id, currentAppUserID, appUserID))
+            deviceCache.clearCaches(oldAppUserID: currentAppUserID, andSaveWithNewUserID: appUserID)
+            completion(nil)
+        }
+    }
+
+    @objc(createAliasForAppUserID:completion:)
+    public func createAlias(appUserID alias: String, completion: @escaping (Error?) -> Void) {
+        guard let currentAppUserID = maybeCurrentAppUserID else {
+            Logger.warn(Strings.identity.creating_alias_failed_null_currentappuserid)
+            completion(ErrorUtils.missingAppUserIDError())
+            return
+        }
+
+        Logger.user(String(format: Strings.identity.creating_alias, currentAppUserID, alias))
```

Comment:
maybe these logs could move to `Backend`?

Reply:
The way this is plumbed, it's difficult to place it right where we do the request. This is because `createAlias` is just a super thin wrapper that ultimately just calls to `performPostRequest`, which then just passes a block to `handle`

I think moving this logging to inside the completion block (just a line or two lower) is a good start. Ultimately, we probably want to refactor the `handle` functions in `Backend.swift` so that we can pass in an optional function that handles logging specific to each request (by checking status codes/errors/whatever). That or maybe we can use aspect oriented programming in Swift now? Is that a thing yet?